src/coreclr/System.Private.CoreLib/src/System/Array.CoreCLR.cs

```diff
@@ -252,8 +268,47 @@ private unsafe nint GetFlattenedIndex(ReadOnlySpan<int> indices)
             }
         }
 
-        [MethodImpl(MethodImplOptions.InternalCall)]
-        internal extern unsafe object? InternalGetValue(nint flattenedIndex);
+        internal unsafe object? InternalGetValue(nint flattenedIndex)
+        {
+            MethodTable* pMethodTable = RuntimeHelpers.GetMethodTable(this);
+
+            TypeHandle arrayElementTypeHandle = pMethodTable->GetArrayElementTypeHandle();
+
+            // Legacy behavior (none of the cases where the element type is a type descriptor are supported).
+            // This includes the case where the element type is a pointer or a function pointer too.
```

Comment:
It is pointer and function pointer cases, and nothing else. This comment makes it sound like there may be some other cases too.

Reply:
Fixed in 23b4a3ffa26e331db50ddce66353e82ce7913722.